web3.js/test/connection.test.js

```diff
@@ -1521,6 +1521,57 @@ test('get largest accounts', async () => {
   expect(largestAccounts.length).toEqual(20);
 });
 
+test('stake activation should throw when called for not delegated account', async () => {
+  const connection = new Connection(url);
+
+  const publicKey = new Account().publicKey;
+  mockRpc.push([
+    url,
+    {
+      method: 'getStakeActivation',
+      params: [publicKey.toBase58()],
+    },
+    {
+      error: {message: 'account not delegated'},
+      result: undefined,
+    },
+  ]);
+
+  await expect(connection.getStakeActivation(publicKey)).rejects.toThrow(
+    `failed to get Stake Activation ${publicKey.toBase58()}: account not delegated`,
+  );
+});
+
+test('stake activation should only accept state with valid string literals', async () => {
+  const connection = new Connection(url);
+
+  const publicKey = new Account().publicKey;
+  const addStakeActivationMock = state => {
+    mockRpc.push([
+      url,
+      {
+        method: 'getStakeActivation',
+        params: [publicKey.toBase58()],
+      },
+      {
+        error: undefined,
+        result: {
+          state: state,
+          active: 0,
+          inactive: 80,
+        },
+      },
+    ]);
+  };
+
+  addStakeActivationMock('active');
+  let activation = await connection.getStakeActivation(publicKey);
```

Comment:
We have mocked out integration tests as well as live integration tests. A bit confusing and redundant in my opinion but the live tests serve as a nice early warning system for breaking changes 😉 

Anyways, this will definitely not pass during a live test. You could set up a delegated stake account ahead of time to ensure this works correctly. It would be nice to check the epoch param as well

Reply:
sure, let me try to figure out how to create delegated stake account that I can use here.

Reply:
thanks for the good advice, running localnet with live tests was helpful.